### PR TITLE
add hyphenation setting to style manager

### PR DESCRIPTION
**Pull request by Fahad-Alsaidi**
part of #190

### CODE REVIEW

**Comment by khaledhosny** [comment]
- Only consecutive hyphenation belongs to paragraph style, the rest belongs to character style.
- No need for a new tab page, it should be a new block/section in the main tab page.

**Comment by Fahad-Alsaidi** [comment]
> Only consecutive hyphenation belongs to paragraph style, the rest belongs to character style.

Well, I looked it from different angel. I group all hyphenation setting in one place. This will be easier for user. Otherwise, the end users will have to look into different places to adjust hyphenation which is definitely not good experience. 
Also, the whole concept of hyphenation is related to the paragraph not to the character. So you find other DTP group them in one place under paragraph styles not in character style such as adobe indesign. 
adobe-ind

> No need for a new tab page, it should be a new block/section in the main tab page.

why not new tab? why it should be all settings crowed in one place??

**Comment by khaledhosny** [comment]
I agree that the split is bad, but we want to apply different settings to different languages even in the same paragraph, so having all hyphenation settings grouped in the paragraph style is not ideal. Imaging having mixed English and Malayalam text in one paragraph, for English you want to use “-” as a hyphenation character, but for Malayalam you want to use ZWNJ, but you can’t do that if the setting is in paragraph styles since the whole paragraph will have the same style. That is the reason we moved hyphenation setting to styles in the first place since previously they were applied globally.

The only hyphenation setting currently that does not make much sense as a character style is the consecutive words as usually you want this same value for full paragraphs or even the full documents.

InDesign is definitely wrong, I want to add left and right hyphen minimum as well (After first and After last in the above screenshot) and this is also different from one language to another, the setting for English is not suitable to Polish or Thai or Malayalam.

**Comment by Fahad-Alsaidi** [comment]
If you look it at whole it is the same. In your case, you will have to make two char styles then you have to apply them in each characters you want.
In my proposal you have to create two paragraph styles then apply them each time you want them.

**Comment by khaledhosny** [comment]
You can’t apply to paragraph styles to the different parts of the same paragraph, could you?

**Comment by Fahad-Alsaidi** [comment]
@khaledhosny you are right. I will modified the patch per your feedback.

**Comment by Fahad-Alsaidi** [comment]
updated !

File: scribus/ui/smcstylewidget.ui [hunk 1]
```diff
@@ -427,20 +412,69 @@
...
-          <width>300</width>
-          <height>0</height>
-         </size>
-        </property>
-       </spacer>
+       <layout class="QHBoxLayout" name="horizontalLayout">
+        <item>
+         <widget class="QLabel" name="label">
+          <property name="text">
+           <string>Smallest Word</string>
```
Comment (khaledhosny) [inline scribus/ui/smcstylewidget.ui]: The name is not very clear, since nothing tells that is is about hyphenation. I’d add a new “Hyphenation” group similar to “Advanced Formatting”, since I expect at least two new hyphenation settings.

File: scribus/ui/smcstylewidget.cpp [hunk 2]
```diff
@@ -191,30 +191,30 @@ void SMCStyleWidget::fillColorCombo(ColorList &colors)
...
-	fillLangComboFromList(languageList);
-	fillColorCombo(m_Doc->PageColors);
-	fontFace_->RebuildList(m_Doc);
-
-	connect(m_Doc->scMW(), SIGNAL(UpdateRequest(int)), this , SLOT(handleUpdateRequest(int)));
+	fillLangComboFromList(languageList);
+	fillColorCombo(m_Doc->PageColors);
+	fontFace_->RebuildList(m_Doc);
+
+	connect(m_Doc->scMW(), SIGNAL(UpdateRequest(int)), this , SLOT(handleUpdateRequest(int)));
```
Comment (khaledhosny) [inline scribus/ui/smcstylewidget.cpp:217]: Can you revert these unnecessary whitespace changes?

File: scribus/ui/smcstylewidget.ui (hunk 1 above)
Comment (Fahad-Alsaidi) [inline scribus/ui/smcstylewidget.ui]: good idea.

File: scribus/ui/smcstylewidget.cpp (hunk 2 above)
Comment (Fahad-Alsaidi) [inline scribus/ui/smcstylewidget.cpp:217]: I really tired many times to remove it but I couldn't. I don't see where I can edit and I don't remember I edited this part.

Comment (khaledhosny) [inline scribus/ui/smcstylewidget.cpp:217]: It seems this part had DOS line endings while the rest of file had Unix line endings. This was just broken of course (Scribus, you know), so I think your editor just fixed it which is fine.
